views/policies.py

```diff
@@ -0,0 +1,183 @@
+        self.ui_utils.sleep(1)
+        try:
+            self.web_driver.find_element_by_xpath('//*[@title="Delete this Middleware Server Policy"]').click()
+            self.ui_utils.accept_alert(10)
+        except Exception, e:
+            try:
+                self.web_driver.find_element_by_xpath(
+                    '//*[@title="Policies that belong to Profiles can not be deleted"]')
+                self.web_session.logger.error('Policy "{}" - Policies that belong to Profiles can not be deleted'.format(policy_name))
+                return False
+            except Exception, e:
+                assert False, 'Policy "{}" not found'.format(policy_name)
+
+        assert self.ui_utils.waitForTextOnPage('Policy "{}": Delete successful'.format(policy_name), 10)
+
+        return True
+
+    def assign_middleware_policy_events(self, policy_name, event_type):
+        self.web_session.logger.info('Assign Policy Events: "{}", Policy Name: "{}"'.format(event_type, policy_name))
+
+        assert event_type in self.EVENT_TYPES, 'Unsupposrted event type "{}". Valid Event types: {}'\
+            .format(event_type, self.EVENT_TYPES)
+
+        self.navigate_to_middleware_server_control_policies(policy_name=policy_name)
+
+        self.web_driver.find_element_by_xpath('//*[@title="Configuration"]').click()
+        self.ui_utils.sleep(2)
+        self.web_driver.find_element_by_xpath('//*[@title="Edit this Policy\'s Event assignments"]').click()
+        self.ui_utils.waitForTextOnPage('"{}" Event Assignments'.format(policy_name), 10)
+
+        els = self.web_driver.find_elements_by_xpath(
+            ".//*[contains(text(), 'No') and not (ancestor::*[contains( @ style, 'display: none')])]")
+
+        if event_type == "deployments":
+            self.__assign_middleware_policy_deployment_events__(policy_name, els)
+        elif event_type == "datasources":
+            self.__assign_middleware_policy_datasource_events__(policy_name, els)
+        elif event_type == 'jdbc':
+            self.__assign_middleware_policy_jdbc_events__(policy_name, els)
+
+        # Save the Assignments
+        el = self.web_driver.find_element_by_xpath('//*[@title="Save Changes"]')
+        assert self.ui_utils.wait_until_element_displayed(el, 10)
+        el.click()
+        assert self.ui_utils.waitForTextOnPage('Policy "{}" was saved'.format(policy_name), 15)
+
+    def __assign_middleware_policy_deployment_events__(self, policy_name, els):
+        assert self.ui_utils.isTextOnPage('"{}" Event Assignments'.format(policy_name))
+
+        # EAP Application/Deployments
+        els[84].click()
+        els[85].click()
+        els[86].click()
+        els[87].click()
+
+    def __assign_middleware_policy_datasource_events__(self, policy_name, els):
+        assert self.ui_utils.isTextOnPage('"{}" Event Assignments'.format(policy_name))
+
+        # EAP Datasource
+        els[88].click()
```

Comment:
Will the element always hold the position on HTML?

Reply:
yes, element positioning may change in the future, and in fact the entire workflow will be changing to a wizard type workflow. So, I will refactor completely when that happens.